File: src/main/java/dev/esophose/discordbot/command/arguments/StringArgumentHandler.java
package dev.esophose.discordbot.command.arguments;

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler;
import discord4j.core.object.entity.Guild;
import reactor.core.publisher.Mono;

public class StringArgumentHandler extends DiscordCommandArgumentHandler<String> {

    @Override
    protected Mono<String> handleInternal(Guild guild, String input) {
        if (input.isBlank())
            return Mono.empty();
        return Mono.just(input);
    }

    @Override
    public String getErrorMessage(Guild guild, String input) {
        return "Invalid string, cannot be empty";
    }

    @Override
    public Class<String> getHandledType() {
        return String.class;
    }

}

File: src/main/kotlin/dev/esophose/discordbot/command/arguments/StringArgumentHandler.kt
package dev.esophose.discordbot.command.arguments

import dev.esophose.discordbot.command.DiscordCommandArgumentHandler
import discord4j.core.`object`.entity.Guild
import reactor.core.publisher.Mono
import kotlin.reflect.KClass

class StringArgumentHandler : DiscordCommandArgumentHandler<String>() {

    override val handledType: KClass<String>
        get() = String::class

    override fun handleInternal(guild: Guild, input: String): Mono<String> {
        return if (input.isBlank()) Mono.empty() else Mono.just(input)
    }

    override fun getErrorMessage(guild: Guild, input: String): String {
        return "Invalid string, cannot be empty"
    }

}
